org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java

```diff
@@ -95,8 +99,20 @@ public void onEvent(Event event) throws Exception {
 		String eventName = event.getName();
 
 		if (eventName.equals(Events.ON_CLICK)) {
-			statusPanel.refresh();
-			statusPanel.updateUI();
+    		BusyDialog busyDialog = new BusyDialog();
+            busyDialog.setShadow(false);
+            getParent().insertBefore(busyDialog, getParent().getFirstChild());
+			ServerPushTemplate template = new ServerPushTemplate(AEnv.getDesktop());
+    		ContextRunnable cr = new ContextRunnable() {
+    			@Override
+				protected void doRun() {
+    				refresh(template);
+    				template.execute(() -> {
+    					busyDialog.detach();

```

Comment:
better to use template.executeAsyn instead

Reply:
I copied that line from DashboardRunnable:146

Must I change it there too?

Reply:
ah yes, should change that too.